_posts/2024-09-26-managing-connector-offsets.md

```diff
@@ -0,0 +1,277 @@
+The process is very similar for both Kafka Connect and MirrorMaker, so we'll demonstrate how to do it with Kafka Connect and then explain how the process applies to MirrorMaker.
+
+### Before we begin
+
+If you want to follow along the steps in this blog post you need a Kubernetes cluster containing the Strimzi operator, a Kafka cluster, a Connect cluster, and a running connector.
+First run through the Strimzi [quickstart guide](https://strimzi.io/quickstarts/) to deploy your Strimzi operator and Kafka cluster.
+
+Once you have a Kafka cluster you can deploy Connect and a connector using the following commands:
+1. `kubectl apply -f https://strimzi.io/examples/latest/connect/kafka-connect-build.yaml -n kafka`
+2. `kubectl wait kafkaconnect/my-connect-cluster --for=condition=Ready --timeout=300s -n kafka`
+3. `kubectl annotate kafkaconnect my-connect-cluster strimzi.io/use-connector-resources=true -n kafka`
+4. `kubectl apply -f https://strimzi.io/examples/latest/connect/source-connector.yaml -n kafka`
+5. `kubectl wait kafkaconnector/my-source-connector --for=condition=Ready --timeout=300s -n kafka`
+
+### Listing offsets
+
+To list offsets, edit your `KafkaConnector` resource by running `kubectl edit kafkaconnector my-source-connector -n kafka` and add the configuration to specify where the offsets should be output:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+# ...
+spec:
+  #...
+  listOffsets:
+    toConfigMap:
+      name: my-connector-offsets
+  #...
+```
+
+This tells Strimzi to write the offsets to a ConfigMap called `my-connector-offsets`.
+
+To trigger Strimzi to get the latest offsets, annotate your `KafkaConnector` resource:
+
+```shell
+$ kubectl annotate kafkaconnector my-source-connector strimzi.io/connector-offsets=list -n kafka
+```
+
+After a couple of minutes you should have a new ConfigMap containing the output:
+
+```shell
+$ kubectl get configmap my-connector-offsets -n kafka -oyaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2024-09-26T10:20:15Z"   (1)
+  labels:
+    strimzi.io/cluster: my-connect-cluster
+  name: my-connector-offsets
+  namespace: kafka
+  ownerReferences:    (2)
+  - apiVersion: kafka.strimzi.io/v1beta2
+    blockOwnerDeletion: false
+    controller: false
+    kind: KafkaConnector
+    name: my-source-connector
+    uid: 637e3be7-bd96-43ab-abde-c55b4c4550e0
+  resourceVersion: "66951"
+  uid: 641d60a9-36eb-4f29-9895-8f2c1eb9638e
+data:   (3)
+  offsets.json: |-
+    {
+      "offsets" : [ {
+        "partition" : {
+          "filename" : "/opt/kafka/LICENSE"
+        },
+        "offset" : {
+          "position" : 15295
+        }
+      } ]
+    }
+```
+
+1. If the ConfigMap doesn't already exist, Strimzi will create it automatically.
+2. The owner reference points to your `KafkaConnector` resource. To provide a custom owner reference, create the ConfigMap in advance and set an owner reference manually.
```

Comment:
"To provide a custom owner reference, create the ConfigMap in advance and set an owner reference manually.", what's the use case for someone doing this?

Reply:
If they don't want the ConfigMap to be cleaned up automatically when they delete the KafkaConnector resource

Reply:
Good point. Maybe adding this "tip" to the line?